dist/profile/manifests/buildagent.pp

```diff
@@ -67,6 +72,46 @@
         'zip',
     ])
 
+    # There is no linux_aarch64 azcopy release, considering that aarch64 = amd64 so vagrant can run on Mac Silicon
+    if  $facts['os']['architecture'] == 'aarch64' {
+      $architecture = 'arm64'
+    } else {
+      $architecture =  $facts['os']['architecture']
+    }
+    $azcopy_url = "https://azcopyvnext.azureedge.net/releases/release-10.21.0-20230928/azcopy_linux_${architecture}_10.21.0.tar.gz"
+    notify { "azcopy_url: ${azcopy_url}": }
+
+    exec { 'Install azcopy':
+      require => [Package['curl'], Package['tar'], Account[$user]],
+      command => "/usr/bin/mkdir -p /tmp/azcopy && /usr/bin/curl ${azcopy_url} | /usr/bin/tar -xz --strip-components=1 -C /tmp/azcopy && /usr/bin/cp /tmp/azcopy/azcopy /usr/local/bin/azcopy && /usr/bin/chmod +x /usr/local/bin/azcopy && /usr/bin/rm -rf /tmp/azcopy/",
```

Comment:
```suggestion
      command => "/usr/bin/curl --silent --location ${azcopy_url} | /usr/bin/tar --extract --gzip --directory=/usr/local/bin/ --strip-components=1 --wildcards '*/azcopy'",
```

Reply:
I tried something like that previously but didn't got it working, thanks!

Addressed in https://github.com/jenkins-infra/jenkins-infra/pull/3099/commits/2586af2f871a3f72edf18f2c00ed03055ff7fc90